t.in.eoarchive.py

```diff
@@ -0,0 +1,612 @@
+                                                  MB or GB
+
+    """
+    # use psutil cause of alpine busybox free version for RAM/SWAP usage
+    mem_available = psutil.virtual_memory().available
+    swap_free = psutil.swap_memory().free
+    memory_GB = (mem_available + swap_free) / 1024.0**3
+    memory_MB = (mem_available + swap_free) / 1024.0**2
+
+    if unit == "MB":
+        memory_MB_percent = memory_MB * percent / 100.0
+        return int(round(memory_MB_percent))
+    elif unit == "GB":
+        memory_GB_percent = memory_GB * percent / 100.0
+        return int(round(memory_GB_percent))
+    else:
+        grass.fatal(_(f"Memory unit <{unit}> not supported"))
+
+
+def browse_eolab_collection(collection, bands, mountpoint, start_date, end_date):
+    """ Browses an EOLAB collection based on the user defined start and end date
+        as well as the current computational region
+    Args:
+        collection(string): user defined collection
+        bands(list): user defined bands
+        mountpoint(string): mountpoint where collection is available
+        start_date(datetime): start date as datetime object
+        end_date(datetime): end date as datetime object
+    Returns:
+        scenes_to_import(list): list of scenes to import. Each list item is a
+                                dictionary with required information.
+                                Dictionary key value pairs are:
+                                    - scene(string): scene name
+                                    - path(string): complete path to scene dir
+                                    - datetime(datetime): timestamp of the scene
+                                    - band_paths(dict): Dictionary with
+                                                        key-value pairs of
+                                                        band filenames and
+                                                        corresponding
+                                                        semantic label
+    """
+    scenes_to_import = list()
+
+    collection_dict = eolab_collection_params[collection]
+    basepath = os.path.join(mountpoint, collection_dict["basepath"])
+    file_format = collection_dict["file_format"]
+    clm_dir = collection_dict["CLM_dir"]
+    tile_system = collection_dict["tile_system"]
+    if tile_system == "UTM":
+        tiles = get_utmcells()
+    bands_filesuffixes = {}
+    for key, value in collection_dict["bands_filesuffixes"].items():
+        if key in bands:
+            # we need them swapped later
+            bands_filesuffixes[value] = key
+
+    for year in os.listdir(basepath):
+        for month in os.listdir(os.path.join(basepath, year)):
+            for day in os.listdir(os.path.join(basepath, year, month)):
+                folder_date = date(int(year), int(month), int(day))
+                if folder_date >= start_date and folder_date <= end_date:
```

Comment:
From https://processes.openeo.org/#load_collection
start_time: The specified instance in time is included in the interval.
end_time: The specified instance in time is excluded from the interval.

Reply:
This has been adapted

Reply:
Now what about start_date = end_date ? In this case folder_date <= end_date should be ok, otherwise you won't get any scenes.

Reply:
Yes, internally end_date is now incremented by 1 day if start_date == end_date